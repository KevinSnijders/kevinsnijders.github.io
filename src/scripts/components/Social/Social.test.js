import React from 'react';
import {
  setUpComponent,
  testComponentMatchSnapshot,
  testComponentIsRendered
} from '../../utils/index';
import Social from './Social';

const testComponent = 'SocialComponent';
const props = [
  {
    icon: 'github-square-brands',
    link: '#'
  },
  {
    icon: 'linkedin-brands',
    link: '#'
  }
];

describe('Social Component', () => {
  describe('Match Snapshot', () => {
    testComponentMatchSnapshot(Social);
  });
  describe('Social icons with state', () => {
    let component;
    beforeEach(() => {
      component = setUpComponent(Social);
      component.setState({
        socials: props
      });
    });
    it('Should render socials', () => {
      testComponentIsRendered(component, testComponent, true);
    });
  });
  describe('Social icons without state', () => {
    let component;
    beforeEach(() => {
      component = setUpComponent(Social);
      component.setState({
        socials: []
      });
    });
    it('Should not render socials', () => {
      testComponentIsRendered(component, testComponent, false);
    });
  });
});
