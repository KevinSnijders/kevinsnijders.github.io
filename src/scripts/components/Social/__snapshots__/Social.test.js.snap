// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<ul
  className="d-flex flex-row "
  data-test="SocialComponent"
>
  <li
    className="pl-2 pr-2"
  >
    <a
      aria-label="display a brand icon"
      href="#"
    >
      <svg
        className="icon icon-github-square-brands "
        data-test="IconComponent"
      >
        <use
          xlinkHref="#icon-github-square-brands"
        />
      </svg>
    </a>
  </li>
  <li
    className="pl-2 pr-2"
  >
    <a
      aria-label="display a brand icon"
      href="#"
    >
      <svg
        className="icon icon-linkedin-brands "
        data-test="IconComponent"
      >
        <use
          xlinkHref="#icon-linkedin-brands"
        />
      </svg>
    </a>
  </li>
</ul>
`;
